### PR TITLE
Bugfix/arsn 426 mpu part upload

### PR DESCRIPTION
**Pull request by benzekrimaha**
Linked issue : https://scality.atlassian.net/browse/ARSN-426

### CODE REVIEW

File: lib/storage/data/external/AwsClient.js [hunk 1]
```diff
@@ -317,6 +317,8 @@ class AwsClient {
     uploadPart(request, streamingV4Params, stream, size, key, uploadId,
         partNumber, bucketName, log, callback) {
         let hashedStream = stream;
+        let callbackCalled = false;
+
```
Comment (ghost) [inline lib/storage/data/external/AwsClient.js]: I think we have a utility for that, `jsutil.once`

File: lib/storage/data/external/AwsClient.js [hunk 2]
```diff
@@ -330,10 +332,15 @@ class AwsClient {
...
+            if (callbackCalled) {
+                return;
+            }
+            callbackCalled = true;
+
             if (err) {
                 logHelper(log, 'error', 'err from data backend ' +
                     'on uploadPart', err, this._dataStoreName, this.clientType);
-                return callback(errors.ServiceUnavailable
+                callback(errors.ServiceUnavailable
```
Comment (ghost) [inline lib/storage/data/external/AwsClient.js]: why removing the return? we will end up with 2 callbacks being called, no?

File: lib/storage/data/external/AwsClient.js (hunk 1 above)
Comment (benzekrimaha) [reply lib/storage/data/external/AwsClient.js]: done here : https://github.com/scality/Arsenal/pull/2254/files#diff-d75a9a19ab15f67df9624710374912016460445162dff59507eba0a27dbbdec2R8

File: lib/storage/data/external/AwsClient.js (hunk 2 above)
Comment (benzekrimaha) [reply lib/storage/data/external/AwsClient.js]: fixed here : https://github.com/scality/Arsenal/pull/2254/files#diff-b50ec864af3d183d79584937afbb4c703a1770dec282faf0b3db89260db4c37cR339

**Review by welansari — APPROVED** [review]

**Review by ghost — APPROVED** [review]

**Pull request merged**